psi4/src/psi4/libmints/integral.cc

```diff
@@ -203,65 +201,93 @@ OneBodyAOInt* IntegralFactory::electric_field(int deriv) {
     return new ElectricFieldInt(spherical_transforms_, bs1_, bs2_, deriv);
 }
 
-TwoBodyAOInt* IntegralFactory::erd_eri(int deriv, bool use_shell_pairs) {
+TwoBodyAOInt* IntegralFactory::erd_eri(int deriv, bool use_shell_pairs, bool needs_exchange) {
     auto integral_package = Process::environment.options.get_str("INTEGRAL_PACKAGE");
+    auto threshold = Process::environment.options.get_double("INTS_TOLERANCE");
 #ifdef USING_simint
-    if (deriv == 0 && integral_package == "SIMINT") return new SimintERI(this, deriv, use_shell_pairs);
+    if (deriv == 0 && integral_package == "SIMINT") return new SimintERI(this, deriv, use_shell_pairs, needs_exchange);
 #endif
+    if (integral_package == "LIBINT2") return new Libint2ERI(this, threshold, deriv, use_shell_pairs, needs_exchange);
 #ifdef USING_erd
     if (deriv == 0 && integral_package == "ERD") return new ERDERI(this, deriv, use_shell_pairs);
 #endif
-    if (deriv > 0 && integral_package != "LIBINT")
+    if (deriv > 0 && integral_package != "LIBINT1")
         outfile->Printf("ERI derivative integrals only available using Libint");
     if (integral_package == "SIMINT" || integral_package == "ERD")
         outfile->Printf("Chosen integral package " + integral_package +
                         " unavailable.\nRecompile with the appropriate option set.\nFalling back to Libint");
+#ifdef ENABLE_Libint1t
     return new ERI(this, deriv, use_shell_pairs);
+#endif
 }
 
-TwoBodyAOInt* IntegralFactory::eri(int deriv, bool use_shell_pairs) {
+TwoBodyAOInt* IntegralFactory::eri(int deriv, bool use_shell_pairs, bool needs_exchange) {
```

Comment:
I think some of these functions need to be cleaned up.

Reply:
was it anything in particular to `needs_exchange` that needed cleaning? I think the follow-up pass to drop ERD and L1 will help here.

Reply:
Yeah, I think you're right that the follow up pass to drop ERD and L1 will do it.